spec/controllers/checkouts_controller_spec.rb

```diff
@@ -19,37 +19,70 @@
     let(:user) do
       { username: 'zzz123',
         name: 'Zeke',
-        patron_key: '1234567' }
+        patron_key: '1234567',
+        session_token: 'e0b5e1a3e86a399112b9eb893daeacfd' }
     end
 
+    let(:mock_patron) { instance_double(Patron, checkouts: checkouts) }
+
+    let(:checkouts) { [
+      instance_double(Checkout, item_key: '123', resource: 'item', bib_summary: 'Renewal 1 (ABC)', due_date: nil),
+      instance_double(Checkout, item_key: '456', resource: 'item', bib_summary: 'Renewal 2 (DEF)', due_date: nil),
+      instance_double(Checkout, item_key: '789', resource: 'item', bib_summary: 'Renewal 3 (GHI)', due_date: nil)
+    ] }
+
     let(:mock_client) do
       instance_double(SymphonyClient, ping?: true)
     end
 
-    let(:checkouts) { [instance_double(Checkout, key: '1', due_date: nil)] }
-
     before do
       warden.set_user(user)
       allow(SymphonyClient).to receive(:new).and_return(mock_client)
       allow(mock_patron).to receive(:checkouts).and_return(checkouts)
     end
 
-    it 'sends the right item details to the web service' do
-      item_details = controller.send(:item_details)
+    describe '#index' do
+      before do
+        allow(ViewCheckoutsJob).to receive(:perform_later)
+      end
 
-      expect(item_details).to eq circRecordList: true
-    end
+      it 'sends a job to ViewCheckoutsJob' do
+        get :index
+
+        expect(ViewCheckoutsJob).to have_received(:perform_later)
+      end
 
-    it 'renders the index template' do
-      get(:index)
+      it 'renders the index template' do
+        get :index
 
-      expect(response).to render_template 'index'
+        expect(response).to render_template 'index'
+      end
     end
 
-    it 'assigns a list of checkouts' do
-      get(:index)
+    describe '#batch_update' do
+      before do
+        allow(RenewCheckoutJob).to receive(:perform_later)
+      end
+
+      it 'requires list of checkouts to be renewed as params' do
+        patch :batch_update, params: {}
+
+        expect(flash[:notice]).to match(/No items were selected/)
+      end
+
+      it 'sends a job to RenewalJob' do
+        patch :batch_update, params: { renewal_list: ['123', '456'] }
+
+        expect(RenewCheckoutJob).to have_received(:perform_later).at_least(2)
```

Comment:
```suggestion
        expect(RenewCheckoutJob).to have_received(:perform_later).twice
```

Should be only twice right? Not 2 or more?

Reply:
yes, twice only, because `renewal_list: ['123', '456']`